### PR TITLE
Add stackage-overlay blog post

### PR DESCRIPTION
**Pull request by 4e6**

### CODE REVIEW

File: posts/2018-01-19-stackage-overlay.markdown [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Yesterday we published a
+[typeable/nixpkgs-stackage](https://github.com/typeable/nixpkgs-stackage)
+Stackage Nixpkgs overlay plus a
+[typeable/stackage2nix](https://github.com/typeable/stackage2nix) update
+with overlay support. Here's a quick overview of new features.
+
+# Stackage overlay
+
+Overlay adds Stakage LTS releases to Nixpkgs at the
```
Comment (dredozubov) [inline posts/2018-01-19-stackage-overlay.markdown]: s/Stakage/Stackage/g

File: posts/2018-01-19-stackage-overlay.markdown [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+```
+
+Stackage packages are generated with _stackage2nix_ and will be updated on a
+regular basis.
+
+## stackage2nix build support
+
+Overlay also adds build utils at _pkgs.haskell.packages.stackage.lib_. You can
+create a build derivation from your project's _stack.yaml_ by using
```
Comment (dredozubov) [inline posts/2018-01-19-stackage-overlay.markdown]: s/derivation/derivations/ ?

File: posts/2018-01-19-stackage-overlay.markdown [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+`callStackage2nix` function calls _stackage2nix_ on provided
+path and produces the derivation with Haskell packages for the project.
+[nixpkgs-stackage/examples](https://github.com/typeable/nixpkgs-stackage/tree/master/examples)
+directory contains full source code of the derivations listed above.
+
+# stackage2nix
+
+_stackage2nix_ was also updated to support the overlay. Now by default, it
```
Comment (dredozubov) [inline posts/2018-01-19-stackage-overlay.markdown]: s/overlay/overlays/

File: posts/2018-01-19-stackage-overlay.markdown [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+creates a single _default.nix_ with the override of existing Stackage packages
```
Comment (dredozubov) [inline posts/2018-01-19-stackage-overlay.markdown]: an override of

File: posts/2018-01-19-stackage-overlay.markdown [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+Stackage Nixpkgs overlay plus a
+[typeable/stackage2nix](https://github.com/typeable/stackage2nix) update
+with overlay support. Here's a quick overview of new features.
+
+# Stackage overlay
+
+Overlay adds Stakage LTS releases to Nixpkgs at the
+`pkgs.haskell.packages.stackage`.
+
+Provided Stackage packages can be used the same way as original Hackage
```
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "Provided Stackage packages" -> "Stackage packages provided", IMHO чуть-чуть понятнее. Первый вариант читается как "При условии, что Stackage packages могут использоваться..."

File: posts/2018-01-19-stackage-overlay.markdown [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+packages maintained by Nix. There are some examples from Nixpkgs [Haskell
+manual](https://nixos.org/nixpkgs/manual/#users-guide-to-the-haskell-infrastructure)
+listed below with `haskellPackages` replaced with LTS packages.
```
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "There are some examples from Nixpkgs [Haskell manual](https://nixos.org/nixpkgs/manual/#users-guide-to-the-haskell-infrastructure) listed below with `haskellPackages` replaced with LTS packages."
->
"Below we present some examples from Nixpkgs [Haskell manual](https://nixos.org/nixpkgs/manual/#users-guide-to-the-haskell-infrastructure) replacing `haskellPackages` with  LTS packages."

File: posts/2018-01-19-stackage-overlay.markdown [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+List packages:
```
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: List packages in LTS 10.0

File: posts/2018-01-19-stackage-overlay.markdown (hunk 2 above)
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "by using" &mdash; "by" не нужен.

File: posts/2018-01-19-stackage-overlay.markdown [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    overrides = self: super: {
+      stackage2nix = haskell.lib.disableSharedExecutables super.stackage2nix;
+    };
+  };
+in
+  stackage
+```
+
+`callStackage2nix` function calls _stackage2nix_ on provided
+path and produces the derivation with Haskell packages for the project.
```
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "produces the derivation" -> "produces a derivation"

File: posts/2018-01-19-stackage-overlay.markdown (hunk 3 above)
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "Now by default, it creates" -> "Now the default is to create"

File: posts/2018-01-19-stackage-overlay.markdown (hunk 4 above)
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "the override" -> "an override".
"packages set" &mdash; либо "package set", либо просто "packages", без "set".

File: posts/2018-01-19-stackage-overlay.markdown [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+[nixpkgs-stackage/examples](https://github.com/typeable/nixpkgs-stackage/tree/master/examples)
+directory contains full source code of the derivations listed above.
+
+# stackage2nix
+
+_stackage2nix_ was also updated to support the overlay. Now by default, it
+creates a single _default.nix_ with the override of existing Stackage packages
+set. The advantage of this approach is that it does not require additional
+`--all-cabal-hashes` and `--lts-haskell` flags (they only needed for Stackage
```
Comment (gabriel-fallen) [inline posts/2018-01-19-stackage-overlay.markdown]: "they only needed" -> "they are only needed"

File: posts/2018-01-19-stackage-overlay.markdown (hunk 3 above)
Comment (gabriel-fallen) [reply posts/2018-01-19-stackage-overlay.markdown]: Там же "the" &mdash; подразумевается какой-то один определённый оверлей. :)

**Comment by 4e6** [comment]
Thanks, everyone!

**Pull request merged**